From 92e15986a43a8009decffc4d5d290272449487a4 Mon Sep 17 00:00:00 2001
From: Peter O'Gorman <peter@pogma.com>
Date: Thu, 1 May 2008 12:40:24 -0500
Subject: [PATCH] Use AC_CHECK_TOOL for otool and otool64.

* libltdl/m4/libtool.m4 (_LT_REQUIRED_DARWIN_CHECKS): Check.
* libltdl/config/ltmain.m4sh (func_mode_link): Use.
Reported by Richard Purdie <rpurdie@rpsys.net>
---
 ChangeLog                  |    7 +++++++
 libltdl/config/ltmain.m4sh |    6 ++----
 libltdl/m4/libtool.m4      |    6 ++++++
 3 files changed, 15 insertions(+), 4 deletions(-)

Index: libtool-2.2.2/ChangeLog
===================================================================
--- libtool-2.2.2.orig/ChangeLog	2008-05-01 21:34:42.000000000 +0100
+++ libtool-2.2.2/ChangeLog	2008-05-01 21:35:47.000000000 +0100
@@ -1,3 +1,10 @@
+2008-05-01  Peter O'Gorman  <peter@pogma.com>
+
+	Use AC_CHECK_TOOL for otool and otool64.
+	* libltdl/m4/libtool.m4 (_LT_REQUIRED_DARWIN_CHECKS): Check.
+	* libltdl/config/ltmain.m4sh (func_mode_link): Use.
+	Reported by Richard Purdie <rpurdie@rpsys.net>
+
 2008-04-01  Gary V. Vaughan  <gary@gnu.arg>
 
 	GNU Libtool 2.2.2 was released.
Index: libtool-2.2.2/libltdl/config/ltmain.m4sh
===================================================================
--- libtool-2.2.2.orig/libltdl/config/ltmain.m4sh	2008-05-01 21:34:42.000000000 +0100
+++ libtool-2.2.2/libltdl/config/ltmain.m4sh	2008-05-01 21:34:48.000000000 +0100
@@ -4961,11 +4961,9 @@
 		    done
 		    if test -f "$absdir/$objdir/$depdepl" ; then
 		      depdepl="$absdir/$objdir/$depdepl"
-		      darwin_install_name=`otool -L $depdepl | $SED -n -e '3q;2,2p' | $SED -e 's/(.*//'`
-		      darwin_install_name=`$ECHO $darwin_install_name`
+		      darwin_install_name=`${OTOOL} -L $depdepl | awk '{if (NR == 2) {print $1;exit}}`
                       if test -z "$darwin_install_name"; then
-                          darwin_install_name=`otool64 -L $depdepl | $SED -n -e '3q;2,2p' | $SED -e 's/(.*//'`
-                          darwin_install_name=`$ECHO $darwin_install_name`
+                          darwin_install_name=`${OTOOL64} -L $depdepl  | awk '{if (NR == 2) {print $1;exit}}`
                       fi
 		      compiler_flags="$compiler_flags ${wl}-dylib_file ${wl}${darwin_install_name}:${depdepl}"
 		      linker_flags="$linker_flags -dylib_file ${darwin_install_name}:${depdepl}"
Index: libtool-2.2.2/libltdl/m4/libtool.m4
===================================================================
--- libtool-2.2.2.orig/libltdl/m4/libtool.m4	2008-05-01 21:34:42.000000000 +0100
+++ libtool-2.2.2/libltdl/m4/libtool.m4	2008-05-01 21:36:48.000000000 +0100
@@ -893,10 +893,16 @@
     rhapsody* | darwin*)
     AC_CHECK_TOOL([DSYMUTIL], [dsymutil], [:])
     AC_CHECK_TOOL([NMEDIT], [nmedit], [:])
+    AC_CHECK_TOOL([OTOOL], [otool], [:])
+    AC_CHECK_TOOL([OTOOL64], [otool64], [:])
     _LT_DECL([], [DSYMUTIL], [1],
       [Tool to manipulate archived DWARF debug symbol files on Mac OS X])
     _LT_DECL([], [NMEDIT], [1],
       [Tool to change global to local symbols on Mac OS X])
+    _LT_DECL([], [OTOOL], [1],
+      [ldd/readelf like tool for Mach-O binaries on Mac OS X])
+    _LT_DECL([], [OTOOL64], [1],
+      [ldd/readelf like tool for 64 bit Mach-O binaries on Mac OS X 10.4])
 
     AC_CACHE_CHECK([for -single_module linker flag],[lt_cv_apple_cc_single_mod],
       [lt_cv_apple_cc_single_mod=no
